pkg/endpoint/bpf.go

```diff
@@ -514,8 +519,16 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 	createdPolicyMap := false
 
 	defer func() {
-		if err != nil {
-			e.Mutex.Lock()
+		if reterr != nil {
+			lockerr = e.LockAlive()
```

Comment:
This must be unconditional as we otherwise leak resources that have already created.